gateway/ln-gateway/src/rpc/rpc_client.rs

```diff
@@ -37,8 +38,16 @@ impl GatewayRpcClient {
         GatewayRpcClient::new(self.base_url.clone(), password)
     }
 
-    pub async fn get_info(&self) -> GatewayRpcResult<GatewayInfo> {
-        let url = self.base_url.join("/info").expect("invalid base url");
+    pub async fn get_info(
+        &self,
+        federation_id: Option<FederationId>,
+    ) -> GatewayRpcResult<GatewayInfo> {
+        let path = match federation_id {
+            Some(id) => format!("/info?federation_id={id}"),
+            None => "/info".to_string(),
+        };
```

Comment:
Nit: string interpolation is prone to injection attacks (wrong/missing escaping), probably ok here because the tool is only used by the admin and not malicious 3rd parties.

Reply:
I wasn't fully aware of it, thanks for the comment. The solution for it is actually pretty straightforward I think. Instead of using query parameters, it could be checked based on the payload `federation_id` field (in order to still use the same route).